straxen/plugins/event_w_bayes_class.py

```diff
@@ -0,0 +1,55 @@
+import strax
+import straxen
+import numpy as np
+
+
+class EventwBayesClass(strax.Plugin):
+    """
+    Append at event level the posterior probability for an
+    S1, S2, alt_S1 and alt_S2
+    """
+    provides = 'event_w_bayes_class'
+    depends_on = ('peaklet_classification_bayes', 'event_basics')
+    data_kind = 'events'
+    __version__ = '0.0.1'
+
+    def infer_dtype(self):
+
+        dtype = []
+        dtype += strax.time_fields
+        for name in ['s1', 's2', 'alt_s1', 'alt_s2']:
+            dtype += [(f'{name}_s1_ln_prob', np.float32, f'Given an {name}, s1 ln probability')]
+            dtype += [(f'{name}_s2_ln_prob', np.float32, f'Given an {name}, s2 ln probability')]
+
+        return dtype
+
+    def compute(self, peaklets, events):
```

Comment:
If you were to do this on peaks (see earlier comment), it would be much faster since you could just select peaks on their start and endtime

Reply:
done